nixos/lib/systemd-lib.nix

```diff
@@ -156,6 +160,23 @@ in rec {
     optional (attr ? ${name} && !(((isInt attr.${name} || isFloat attr.${name}) && min <= attr.${name} && max >= attr.${name}) || elem attr.${name} values))
       "Systemd ${group} field `${name}' is not a value in range [${toString min},${toString max}], or one of ${toString values}";
 
+  assertRangeWithOptionalMask = name: min: max: group: attr:
+    if (attr ? ${name}) then
+      if isInt attr.${name} then
+        assertRange name min max group attr
+      else if isString attr.${name} then
+        let
+          fields = match "([0-9]+)(/([0-9]+))?" attr.${name};
```

Comment:
Could this be made to also support the hex notation? These can be really handy when specifying bitmasks, and is supported by systemd